### PR TITLE
[Merged by Bors] - feat: Make the coercion `ℝ≥0 → ℝ≥0∞` commute defeqly with `nsmul` and `pow`

### PR DESCRIPTION
**Pull request by YaelDillies**
by tweaking the definition of the `AddMonoid` and `MonoidWithZero` instances for `WithTop`. Also unprotect `ENNReal.coe_injective` and rename `ENNReal.coe_eq_coe → ENNReal.coe_inj`.

From LeanAPAP

---

- [x] depends on: #10275

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by eric-wieser — APPROVED** [review]
bors d+

Thanks! CI is unhappy, but this should probably be an easy fix.

Is there a similar fix to be made to `npow`?

File: Mathlib/Data/ENNReal/Basic.lean [hunk 1]
```diff
@@ -176,7 +177,12 @@ instance canLift : CanLift ℝ≥0∞ ℝ≥0 ofNNReal (· ≠ ∞) := WithTop.c
 
 @[simp] theorem some_eq_coe' (a : ℝ≥0) : (WithTop.some a : ℝ≥0∞) = (↑a : ℝ≥0∞) := rfl
 
-protected theorem coe_injective : Function.Injective ((↑) : ℝ≥0 → ℝ≥0∞) := WithTop.coe_injective
+lemma coe_injective : Injective ((↑) : ℝ≥0 → ℝ≥0∞) := WithTop.coe_injective
+
+@[simp, norm_cast] lemma coe_inj : (p : ℝ≥0∞) = q ↔ p = q := coe_injective.eq_iff
```
Comment (eric-wieser) [inline Mathlib/Data/ENNReal/Basic.lean:182]: Please mention this rename in the commit message.

**Comment by eric-wieser** [comment]
> Also reorder `WithOne` material. `Algebra.Group.WithOne.Defs` was hiding a `Ring` import! I credit Johan and Mario for [leanprover-community/mathlib#2707](https://github.com/leanprover-community/mathlib/pull/2707).

This PR is getting quite big; can you do this separately?

**Comment by YaelDillies** [comment]
bors merge

**Pull request closed without merging**